src/views/bonds/AllBonds.js

```diff
@@ -1,7 +1,44 @@
-import React from 'react'
+import { cilPlus } from '@coreui/icons'
+import CIcon from '@coreui/icons-react'
+import { CCard, CCardBody, CCardHeader, CCol, CRow, CButton } from '@coreui/react'
+import React, { useEffect, useState } from 'react'
+import { axiosPrivate } from 'src/utils/axiosPrivate'
+import BondsTable from './BondsTable'
 
 const Bonds = () => {
-  return <></>
+  const [bonds, setBonds] = useState({})
+
+  const fetchBonds = () => {
+    axiosPrivate.get('v1/bond/').then((response) => {
+      setBonds(response.data)
+    })
+  }
+
+  useEffect(() => {
+    fetchBonds()
+  }, [setBonds])
```

Comment:
Why do we have the setter function (setBonds) as a dependency? Wouldn't bonds be more appropriate? The component only rerenders when the value of bonds changes, right?

Reply:
when we have bonds as a dependency, there will be an infinite loop, because fetchBonds calls setBonds, which changes the state of the bonds. however, when we have setBonds as the dependency, the bonds displayed will only re render if we call setBonds.

Reply:
@urandu We can safely remove `setBonds` from the dependency array. It offers no benefit in this context. @leroy-netizen note that this sometimes is a necessary pattern in those rare scenarios where a function like `setBonds` is memoized and `fetchBonds` calls `setBonds` somewhere in its scope.

Reply:
We shall be calling setBonds when we create or edit a bond on the table, so I will check it in as is, it will utilise it in a subsequent PR with CRUD functionality.